be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -367,6 +367,12 @@ Status SegmentIterator::_get_row_ranges_from_conditions(RowRanges* condition_row
     RowRanges zone_map_row_ranges = RowRanges::create_single(num_rows());
     // second filter data by zone map
     for (auto& cid : cids) {
+        if (_inverted_index_iterators[_schema.unique_id(cid)] != nullptr &&
```

Comment:
may be we can use the same logic above to tell whether it's a fulltext match query or not?